activesupport/lib/active_support/testing/assertions.rb

```diff
@@ -30,6 +30,71 @@ def assert_nothing_raised
         yield
       end
 
+      # Test varying numeric difference between the return value of an expression as a
+      # result of what is evaluated in the yielded block.
+      #
+      #   assert_differences ['Article.count'] do
+      #     post :create, params: { article: {...} }
+      #   end
+      #
+      # An arbitrary expression is passed in and evaluated.
+      #
+      #   assert_differences ['Article.last.comments(:reload).size'] do
+      #     post :create, params: { comment: {...} }
+      #   end
+      #
+      # An arbitrary positive or negative difference can be specified.
+      # The default is <tt>1</tt>.
+      #
+      #   assert_differences ['Article.count', -1] do
+      #     post :delete, params: { id: ... }
+      #   end
+      #
+      # An array of expressions and the predicted change can also be passed in and evaluated.
+      #
+      #   assert_differences [ ['Article.count', 2], ['Post.count', 1] ] do
+      #     post :create, params: { article: {...} }
+      #   end
+      #
+      # A lambda or a list of lambdas can be passed in and evaluated:
+      #
+      #   assert_differences [->{ Article.count }, 2] do
+      #     post :create, params: { article: {...} }
+      #   end
+      #
+      #   assert_differences [ [->{ Article.count }, 2], [->{ Post.count }, 1] ] do
+      #     post :create, params: { article: {...} }
+      #   end
+      #
+      # An error message can be specified.
+      #
+      #   assert_differences ['Article.count', -1], 'An Article should be destroyed' do
+      #     post :delete, params: { id: ... }
+      #   end
+      def assert_differences(expression_array, message = nil, &block)
+        expressions = Array(expression_array)
+        expressions = [expressions] unless expressions.all? { |e| e.kind_of? Array }
```

Comment:
`assert_differences` needs a 2-d array.  This line converts `['Foo']` into `[ ['Foo'] ]` while leaving arrays that are already 2-d untouched